truffle/contracts/VaultNotary/VaultNotary.sol

```diff
@@ -0,0 +1,233 @@
+      * @param vaultId bytes16 The ID of Vault to grant permission
+      * @param anotherAddress address The address to grant permission
+      */
+    function grantUpdateUriPermission(bytes16 vaultId, address anotherAddress)
+        external
+        vaultOwnerOnly(vaultId)
+    {
+        require(!isNotRegistered(vaultId), "vaultId should be registered before grantUpdateUriPermission");
+        notaryMapping[vaultId].aclUriMapping[anotherAddress] = true;
+        emit UpdateUriPermissionGranted(msg.sender, vaultId, anotherAddress);
+    }
+
+    /** @notice Function to revoke update permission to both the Uri field in one vault
+      * @param vaultId The ID of Vault to revoke permission
+      * @param anotherAddress address The address to revoke permission
+      */
+    function revokeUpdateUriPermission(bytes16 vaultId, address anotherAddress)
+        external
+        vaultOwnerOnly(vaultId)
+    {
+        require(!isNotRegistered(vaultId), "vaultId should be registered before revokeUpdateUriPermission");
+        notaryMapping[vaultId].aclUriMapping[anotherAddress] = false;
+        emit UpdateUriPermissionRevoked(msg.sender, vaultId, anotherAddress);
+    }
+
+    /** @notice This is used in unit tests to verify the values are correct after using the setters
+      * Can also be used for outside users to read the vaultUri and vaultHash
+      * @param vaultId bytes16 The ID of the Vault to query
+      * @return vaultUri string The uri of the vault
+      * @return vaultHash string The hash of the vault
+      */
+    function getVaultNotaryDetails(bytes16 vaultId)
+        external
+        view
+        returns(string memory vaultUri, string memory vaultHash)
+    {
+        return (notaryMapping[vaultId].vaultUri, notaryMapping[vaultId].vaultHash);
+    }
+
+    /** @notice Function to set the vault URI
+      * @param vaultId bytes16 ID of the vault to set
+      * @param vaultUri string The vault URI to set
+      */
+    function setVaultUri(bytes16 vaultId, string memory vaultUri)
+        public
+        whitelistedOnlyForUri(vaultId)
+    {
+        notaryMapping[vaultId].vaultUri = vaultUri;
+        emit VaultUri(msg.sender, vaultId, vaultUri);
+    }
+
+    /** @notice Function to set the vault hash
+      * @param vaultId bytes16 ID of the vault to set
+      * @param vaultHash string The vault hash to set
+      */
+    function setVaultHash(bytes16 vaultId, string memory vaultHash)
+        public
+        whitelistedOnlyForHash(vaultId)
+    {
+        notaryMapping[vaultId].vaultHash = vaultHash;
+        emit VaultHash(msg.sender, vaultId, vaultHash);
+    }
+
+    /** @notice This function is only used for testing whether a Vault has been registered yet
+      * @param vaultId bytes16 The ID of the vault to check
+      * @return isRegistered A boolean, true - not registered; false - registered
+      */
+    function isNotRegistered(bytes16 vaultId)
```

Comment:
This function is not much different than a modifier, especially since it's an Internal View method.  I recommend making this a modifier instead and creating the related `isRegistered` modifier as well.